### PR TITLE
fix: check element for null in case of a disposed player

### PR DESCRIPTION
**Pull request by travisbader**
## Description
https://github.com/videojs/video.js/issues/5706

## Specific Changes proposed
Fail fast

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Review by brandonocasey — COMMENTED** [review]
Should we do this in most of the functions in this file?

**Comment by gkatsev** [comment]
It's possible but generally they expose worse errors which are otherwise hidden by this null check.

**Comment by brandonocasey** [comment]
I wonder if we should log a warning in this case then @gkatsev

File: src/js/utils/dom.js [hunk 1]
```diff
@@ -280,6 +280,10 @@ export function addClass(element, classToAdd) {
  *         The DOM element with class name removed.
  */
 export function removeClass(element, classToRemove) {
+  // Protect in case the player gets disposed
+  if (!element) {
```
Comment (gkatsev) [inline src/js/utils/dom.js:284]: Can we add a log here something like
```
log.warn("removeClass was called with an element that doesn't exist");
```

**Review by gkatsev — COMMENTED** [review]
We spoke about it and while we still have reservations about it, if we add a log for this and do the null check we can merge it in.

**Comment by DaveStein** [comment]
@brandonocasey @gkatsev does this address my errors like this?

```
Uncaught TypeError: Cannot read property 'parentNode' of null
Uncaught TypeError: Cannot read property 'language' of null
Uncaught TypeError: Cannot read property 'classList' of null
Uncaught TypeError: Cannot read property 'classList' of null
Uncaught TypeError: Cannot read property 'classList' of null
```

in minified code I had seen it as coming from this line, when `on` is fired for `ready` event

```
this.readyQueue_=[],e&&e.length>0&&e.forEach((function(e){e.call(this)}),this),this.trigger("ready")}),1)},t.$=function(e,t){return jt(e,t||this.contentEl())},t.$$=function(e,t){return Ft(e,t||this.contentEl())},
```

**Comment by gkatsev** [comment]
@DaveStein could be. Do you happen to have a live test page for this that you can share?

**Comment by DaveStein** [comment]
@gkatsev unfortunately i haven't reproduced this yet myself but just see it in error logs.

**Comment by gkatsev** [comment]
@DaveStein if you try latest `main`, we just merged a fix related to the readyQueue not being cleared out properly. We don't have a release out yet, but hopefully soon.

**Comment by DaveStein** [comment]
Thanks for the update @gkatsev. Since I haven't reproduced this locally, I'll need to wait for the release to be able to update my repo and deploy it. I imagine the worst case scenario for a release update is that my issue won't be fixed. If it does, I'll comment in this thread as well so future people searching Google for the error will see this PR fixes both issues.

**Comment by gkatsev** [comment]
Yes, once we get a release out, please circle back whether it's fixed or not.

**Comment by Szefczuk** [comment]
Hi, when are you planning to merge this pr?

**Comment by damjan25** [comment]
Any plans to merge this in next release ?

File: src/js/utils/dom.js (hunk 1 above)
Comment (gkatsev) [reply src/js/utils/dom.js:284]:
```suggestion
  if (!element) {
    log.warn("removeClass was called with an element that doesn't exist");
```

**Review by gkatsev — APPROVED** [review]

**Review by brandonocasey — APPROVED** [review]

**Pull request merged**